ui/redux/selectors/memberships.js

```diff
@@ -0,0 +1,256 @@
+export const selectPendingBuyMembershipIds = (state: State) => selectState(state).pendingBuyIds;
+export const selectPendingCancelMembershipIds = (state: State) => selectState(state).pendingCancelIds;
+export const selectChannelMembershipsByCreatorId = (state: State) => selectState(state).channelMembershipsByCreatorId;
+export const selectById = (state: State) => selectState(state).membershipListById || {};
+export const selectDidFetchMembershipsDataById = (state: State) => selectState(state).didFetchMembershipsDataById;
+export const selectMembershipPerks = (state: State) => selectState(state).membershipPerks;
+export const selectMySupportersList = (state: State) => selectState(state).mySupportersList;
+export const selectProtectedContentClaimsById = (state: State) => selectState(state).protectedContentClaimsByCreatorId;
+export const selectIsListingAllMyTiers = (state: State) => selectState(state).listingAllMyTiers;
+
+export const selectMembershipMineFetched = (state: State) => selectMembershipMineData(state) !== undefined;
+
+export const selectMyActiveMembershipsForChannelClaimId = (state: State, id: string) => {
+  const byId = selectMyActiveMembershipsById(state);
+  return byId && byId[id];
+};
+
+export const selectMyValidMembershipsById = createSelector(
+  selectMyPurchasedMembershipsById,
+  (purchasedMembershipsById) => {
+    const validMembershipsById = {};
+
+    for (const creatorChannelId in purchasedMembershipsById) {
+      const purchasedCreatorMemberships = purchasedMembershipsById[creatorChannelId];
+
+      for (const membership of purchasedCreatorMemberships) {
+        if (membership.Subscription.current_period_end * 1000 > Date.now()) {
+          validMembershipsById[creatorChannelId] = new Set(validMembershipsById[creatorChannelId]);
+          validMembershipsById[creatorChannelId].add(membership);
+          // $FlowFixMe
+          validMembershipsById[creatorChannelId] = Array.from(validMembershipsById[creatorChannelId]);
+        }
+      }
+    }
+
+    return validMembershipsById;
+  }
+);
+
+export const selectMyValidMembershipsForCreatorId = (state: State, id: string) =>
+  selectMyValidMembershipsById(state)[id];
+
+export const selectUserHasValidMembershipForCreatorId = (state: State, id: string) => {
+  const validMemberships = selectMyValidMembershipsForCreatorId(state, id);
+  return Boolean(validMemberships && validMemberships.length > 0);
+};
+
+export const selectMyValidMembershipIds = (state: State) => {
+  const validMembershipsById = selectMyValidMembershipsById(state);
+
+  const validMembershipIds = new Set([]);
+  for (const creatorId in validMembershipsById) {
+    const memberships = validMembershipsById[creatorId];
+
+    for (const membership of memberships) {
+      validMembershipIds.add(membership.MembershipDetails.id);
+    }
+  }
+
+  // $FlowFixMe
+  return validMembershipIds.size ? Array.from(validMembershipIds) : null;
```

Comment:
:small_blue_diamond: Just following up with what we discussed in Slack, the above is ok-ish because the Set is created once and re-used (assuming we are trying to avoid duplicates here).

However, the code below is a bit odd ...
```
    for (const creatorChannelId in purchasedMembershipsById) {
      const purchasedCreatorMemberships = purchasedMembershipsById[creatorChannelId];
      for (const membership of purchasedCreatorMemberships) {
        if (membership.Subscription.current_period_end * 1000 > Date.now()) {
          validMembershipsById[creatorChannelId] = new Set(validMembershipsById[creatorChannelId]);
          validMembershipsById[creatorChannelId].add(membership);
          // $FlowFixMe
          validMembershipsById[creatorChannelId] = Array.from(validMembershipsById[creatorChannelId]);
        }
      }
```
... when a simple `includes` check before doing a `push` is just as readable and much more efficient than creating the transient Set on every iteration, and then re-converting back.